### PR TITLE
Add gradle task to run tests in parallel

### PR DESCRIPTION
**Pull request by rmoreliovlabs**
## Description
* Add gradle task to run tests in parallel
* Updated tests in order to allow them run in parallel

## Motivation and Context
Even though our tests are not properly designed to run in parallel, sometimes it’s worth doing that during development to speed the process up. This should be treated as a dev tool and should not be used in our ci/cd pipelines.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/build.gradle [hunk 1]
```diff
@@ -27,6 +27,21 @@ jacocoTestReport {
     }
 }
 
+task quickTest(type: Test) {
```
Comment (asoto-iov) [inline rskj-core/build.gradle:30]: Maybe we can update the name and use parallelTest instead of quick.

**Comment by asoto-iov** [comment]
Additionally I would add a warning message about the usage of parallel testing must not be done in our CI/CD scripts (at least for now) and also a warning about tests using shared resources could fail.

File: rskj-core/build.gradle [hunk 2]
```diff
@@ -27,6 +27,21 @@ jacocoTestReport {
...
+    useJUnitPlatform()
+    //increase heap memory
+    jvmArgs '-Xss64m', '-Xmx3G'
+    //stop build after the first failing test
+    failFast = true
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/build.gradle]: Addressed

File: rskj-core/build.gradle (hunk 1 above)
Comment (rmoreliovlabs) [reply rskj-core/build.gradle]: Addressed

**Review by asoto-iov — APPROVED** [review]

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Pull request merged**